salt/_modules/metalk8s.py

```diff
@@ -5,6 +5,7 @@
 import logging
 import os.path
 import re
+import six
```

Comment:
If we're moving to Python 3, why using the `six` compat layer?

Like:
> - replace `basestring` with `six.string_types`

Why not using `str` directly?

Otherwise we're gonna have two migration PR?
2only -> 2-3 compat and then 2-3compat -> 3 only?

Reply:
It's more because for the moment we cannot rewrite the salt module unit test to work in python3, and we will do it in the future.
Also, I didn't tested yet but for upgrade purpose, we will maybe need to have a module compatible with both py2 and py3